### PR TITLE
IPC balance changes

### PR DESCRIPTION
**Pull request by Ziiro**
🆑 
balance: IPCs now take less damage from EMPs
balance: EMPs now stun and dismember IPCs
balance: Ultra-lube and Surge have been removed
/ 🆑 

Why: 
* Incidental EMPs like from a singulo or occupational hazards are no longer lethal
* Antags that use EMPs can still kill IPCs fairly easily. Just not with one click for free. It's a long stun and it dismembers them. (Two EMPs will still likely kill them)
* Ultra-lube and Surge fill an unnecessary role. IPCs are immune to the bad chems, why should they get good ones? You can't have your cake and eat it too.

### CODE REVIEW

**Comment by KasparoVy** [comment]
>You can't have your cake and eat it too.
Will be determined by whoever's calling the shots nowadays.

**Comment by shazbot194** [comment]
Sad to see Ultra-lube go, but I will happily trade it for less EMP damage, although I think the dismemberment is a little overkill since it still only takes 3-4 EMPs to dismember them completely.

**Comment by Spacemanspark** [comment]
Sure.

**Comment by davidchan13** [comment]
>IPCs are immune to the bad chems, why should they get good ones? You can't have your cake and eat it too.

When IPCs are immune polyacid and CFL3, till then these no metab meme needs to die.

File: code/modules/surgery/organs/organ.dm [hunk 1]
```diff
@@ -265,12 +265,18 @@
...
 					owner.Stun(5)
-	else
+	else // IPC Robotic Limbs
 		switch(severity)
 			if(1)
-				receive_damage(0, 20)
+				receive_damage(5, 10)
+				owner.Stun(30)
+				owner.AdjustSilence(30)
+				if(src.body_part != UPPER_TORSO && src.body_part != LOWER_TORSO && prob(15)) // Rolled on each EMP'd limb, except for the torso.
```
Comment (tigercat2000) [inline code/modules/surgery/organs/organ.dm]: Don't use `src.`, it's automatically implied.

File: code/modules/surgery/organs/organ.dm [hunk 2]
```diff
@@ -265,12 +265,18 @@
...
+					src.droplimb(0, DROPLIMB_BLUNT) // Blows them off.
```
Comment (tigercat2000) [inline code/modules/surgery/organs/organ.dm]: Don't use `src.`, it's automatically implied.

**Comment by FalseIncarnate** [comment]
I find it adorable you consider Ultra-lube and Surge as "good" chems. They are incredibly punishing.

As for the EMP changes, I cannot support them as you present them. EMPs are the main weakness of the race, and should not be downplayed. Very few non-antag sources of EMPs exist, and are entirely avoidable with even a modicum of forethought. Singulo EMPs are on a predictable interval, and you can easily avoid them by using the camera monitor in the hall to monitor a growing EMP without endangering yourself, then ducking into the control room and slapping the PA down to 0 between pulses with little challenge. Chemical reaction EMPs are a deliberate matter, you can avoid them by not mixing iron and uranium. Similarly, you can avoid the EXPERIMENTOR EMP by (and this is gonna be a real shocker) not using the machine which is entirely optional and frankly needs to be made more interesting mechanically anyways.

You want to essentially turn IPCs into an even more powerful race than they are. They frankly are one of our best balanced karma races, they don't need this buff. Try as you might to claim this isn't one, but it clearly is a buff when you downplay their single largest weakness into an inconvenience.

**Comment by Ziiro** [comment]
Addressing a couple things.

>When IPCs are immune polyacid and CFL3, till then these no metab meme needs to die.

That's not the type of immunity I meant. I meant to things like sleepy pens, neuro-toxin and the like. Immune to airborne chems that fuck up literally every other species on the station. IPCs can still be injected with acids, *because that just makes sense that acids destroy their components and do burn damage.* Giving IPCs a krokodil(surge) and meth(ultralube) equivalent is just unnecessary and partially defeats the purpose of what their differences in chem processing *should* mean.

With regards to the actual effects of the change:



They die in two blinks now, instead of one. Ion rifles can be used to stun them, and nonlethally capture IPC antags as well instead of, you know, murdering them. The damage applied/stun is still more than enough to kill an IPC if you put any other effort into it. I mean, shit. Apply EMP, apply toolbox. Dead IPC that couldn't defend itself. EMP screech, armblade, done. Etc etc.

I'd like to outline it in a sentence here: **Yes, EMPs from antags should still lead to the deaths of IPCs, but it should not be a free, somewhat stealthy kill.**

**Comment by taukausanake** [comment]
From what I understand, IPCs are the _most_ balanced race in the game right now and that's saying something. Really shouldn't be trying to change them up if there's nothing from with them

**Comment by davidchan13** [comment]
You do realize these chems were given to IPCs because they are the only reliable stun negation IPCs have access to, and come with a steep cost to use unlike every other stun resist chem. I understand why people don't like them in the game, but they serve a very specific purpose for balance and taking them away because 'NO CHEMS' is stupid for the same reason giving IPCs actual immunity to all chems would be stupid. I don't see anyone taking healing meds away from plasmemes even though they have all the same immunities you just listed.

**Comment by Dinarzad** [comment]
> You want to essentially turn IPCs into an even more powerful race than they are. They frankly are one of our best balanced karma races, they don't need this buff. Try as you might to claim this isn't one, but it clearly is a buff when you downplay their single largest weakness into an inconvenience.

My sides are roughly orbiting Pluto by now.

When you have half the server telling you "They need a little love, and one-shot kill mechanics are simultaneously, stupid, not fun and a terrible way to balance something" it MIGHT actually be a thought worth entertaining, it gets said so much for a *reason*.
'Turning their weakness into an inconvenience. Like... Oh yeah. It's totally just an inconvenience to have all of your limbs explode off your body and being stunned for a huge length of time is just that sort of minor problem you get back up from, brush off and go "Wow, that was an event, better get back to work". It's not like even a single EMP is opening you up for a murder spree or an incidental non-antag EMP might not leave you stranded long enough to be hit by another one or for some shithead to come rob you or anything.
Like are you *actually* reading what the changes are? THIS is what we're calling an "Inconvenience" these days?

**Comment by FalseIncarnate** [comment]
@Dinarzad 15% chance for any given limb to pop off. This is not guaranteed to pop ANY limbs. And this is the STRONG version of the EMP effect. Additionally, this same level of EMP used to do 20 burn damage, but this change makes it 5 brute (I'm really not sure why it would even do essentially blunt force damage from an EMP) and 8 burn, meaning they now are only doing 65% of their previous damage, split between two damage types for no discernible reason, just to add a stun.

The weaker version of the EMP at least does equivalent damage (was 7 burn, now 2 brute and 5 burn. again, why the split?).

Do note, these damage numbers are BEFORE the additional 50% damage received by IPCs from brute and burn damage types, so 19.5 (split unevenly between 2 types) vs 30 (one type) in strong, equal still on weaker EMP (10.5).

You also are claiming that half the server is saying something, which if true would still mean very little in terms of how balanced they actually are and whether or not this is the way to address those issues. If your concern is with one-hit kills being too common, perhaps look at the sources of those kills, rather than their victims.

And, just for the sake of humoring me, @Ziiro : why would you want to use the ion rifle to non-lethally capture an IPC when your taser, stunbaton, rubber bullets, and disabler beams all work exactly the same on them as they do on a human? That's not what the ion rifle is for, since there are already plenty of non-lethal takedowns that work on them.

**Comment by Tayyyyyyy** [comment]
EMPs are an easy source of AOE mass murder. What I like about this PR is that it makes EMPs just as effective of an antag weapon (there's no way you're getting out of being EMPed alive if an antag is intentionally targeting you) but you're less likely to be collateral damage.

@FalseIncarnate the other examples you give all must be aimed at a single target. EMPs allow you to sort of just wave your hand in the general direction of a bunch of IPCs and guarantee their doom.

**Comment by Ziiro** [comment]
>@Ziiro : why would you want to use the ion rifle to non-lethally capture an IPC when your taser, stunbaton, rubber bullets, and disabler beams all work exactly the same on them as they do on a human? 

AOE. One shot. Click near them, and you'll get them. Also it's a ranged option you can use on an IPC with carp (provided you hit the wall/floor around them without hitting them directly)

**Comment by shazbot194** [comment]
Imagine if we let any traitor buy a massively infections disease grenade for 5 TC, but it cannot be spread from person to person, only from grenade to person, and killed in 30 seconds with an infection radius of 10 tiles. Does that sound like at all fun for anyone but the traitor? Because that would be way less lethal to a room of humans then an EMP is to a room of IPCs. As it isn't instant death, it has a limited range, costs 2.5x more, you can wear protection from it, and cannot penetrate walls.

**Comment by SilvernKell** [comment]
The idea that IPCs are "the most balanced race in the game right now" seems kind of absurd, any change like this that makes it less likely to be random collateral is extremely warranted but I do agree the split between brute and burn damage is entirely unecessary..

**Comment by FalseIncarnate** [comment]
I will admit the sound of "IPCs are the most balanced race" does sound absurd, but when you look at their pros, cons, and neutral-ish mechanics, its actually quite balanced.

- Immunity to 2 major damage types (toxin and oxy), plus 2 minor types (rad and clone) in exchange for a massive vulnerability to EMPs and being emagged
- Immune to viruses (this is both a pro and a con in one, as they can't benefit from them either)
- Does not feel pain and does not breathe in exchange for an increased brute/burn damage received
- Immune to many standard chems (pro and con again, can't benefit from easy healing), while still being susceptible to some nasty ones. Even the synthetic-only chems carry some solid drawbacks.
- Cannot be multiple types of antagonists (I'd call this neutral overall, its not really a boon or detriment to gameplay)
- Can be revived simply by repairing damage directly (no SR or defib needed) in exchange for easily falling apart and requiring surgery to repair
- Cannot receive genetics mutations (calling this one neutral because that includes both powers and disabilities)

For every positive aspect of their mechanics, they have equally powerful downsides. Compared to other races such as Kidan, Vox, or Diona, the IPC tradeoffs are the most balanced (obviously we're ignoring humans here since they have no special aspects).

When you choose to play an IPC, you accept these weaknesses as part of the experience. If you don't want them, it is as simple as selecting another race and just adding robotic limbs. Your species choice should have an impact on your gameplay, lessening that impact by downplaying their weaknesses makes the choice less meaningful and overall less enjoyable.

I routinely see people referencing the traitor EMP kit as their reason for their opinion, which makes me think you'd be better off adjusting the cost of that kit or its contents/how they function. Balancing a race around a limited set of items only makes sense when those items are necessary to play the race (like Vox and plasmaman tanks).

**Comment by davidchan13** [comment]
@FalseIncarnate The Emag thing was removed a long time ago unless someone brought it back recently.

**Comment by Dinarzad** [comment]
> Immunity to 2 major damage types (toxin and oxy), plus 2 minor types (rad and clone, which admittedly are practically non-existent in terms of how frequently they are encountered anyways) in exchange for a massive vulnerability to EMPs and being emagged

This is actually incorrect, IPC **can** accrue toxin damage. It is not as simple, but it can be applied to them, that's the reason "Degreaser" had to be created in the first place, because before IPCs got a toxin-cleaning chemical, if they accrued any, they had it permanently.

> For every positive aspect of their mechanics, they have equally powerful downsides.

This is purely in the realm of the subjective. 
You could ask thirty different people whether those pros/cons are equal and you'd get thirty different answers. A "One shot stealth kill through walls at extreme range" as a con is an objectively terrible design feature, and even if that were kept, the amount of other cons they get on top of it, is positively insane for the double-edged sword of boons they receive.

Meanwhile the Vox are a highly space resistant race, who are also immune to atmos problems unless someone actively strips their mask off and then still take around 7 minutes to have it be a major problem and a smaller brute vulnerability.
Plasmapeople have total space immunity and similar atmos immunity as Vox, just for not being able to swap suits. Their breathing "Con" is even less of a detriment because anyone in position to strip a plasmaman of their plasma tank, can just as easily strip their suit off, which is far more effective of a killer.

IPCs are NOT space immune.
Exchange immunity to breathe damage and resistance to toxin, for a vulnerability to the two most common damage types in the game.
Have their limbs dismembered far easier.
Are vulnerable to 1-click instant kills from range, through walls.
Are "Immune" to soft-crit but if you go into hard crit and don't die, congrats, you better hope someone finds you or you are eternally stuck. and if you ghost out of your body it counts as leaving the round early, not as leaving a dead body.

They gain all those massive detriments for the price of not breathing, not feeling pain and being "Easy" to revive. Those are the only objective 'upgrades' an IPC get that is not itself a double-edged sword of a trait.
You want them to be "Easy to kill, Easy to revive" then cool. I get that. That's why the damage multiplier is there, that's why their limbs pop off like cardboard cut-outs. A One-click kill mechanic is **insane** for any feature, especially one that can be done through walls. This is the ideal compromise. It still is crippling to an EMP, it is still an incredible weakness and vulnerability. It is still a sure-fire way to murder an IPC target as an antag, but it also still means someone has to put in a bare minimum of fucking effort to do it, they still have to put themselves at risk of being seen at least SOMEWHAT. 
That's not asking a whole lot.

**Comment by FalseIncarnate** [comment]
Degreaser (which I made, along with the other synth reagents, so I'm more than familiar with) was made because at the time it was introduced, it was actually ADDING toxins damage (displayed as "residue" on analyzers) to IPCs as something they could accrue. This was since changed, and now degreaser is essentially synthetic Haloperidol with the added benefit of being able to clean lube from floors. It does absolutely nothing to toxin damage, as IPCs aren't meant to get it and any time they do it is due to a bug or oversight that should be reported and can easily be fixed by an admin as IT SHOULDN'T BE HAPPENING. Using a removed usage of a chemical and possible bug/exploit as justification for your argument isn't gonna fly.

We can argue the EMP through walls is an issue in circles until the heat death of the universe, but the fact isn't the vulnerability to the tactic, its the tactic itself. IPCs are especially vulnerable to this simply because all their limbs are robotic, but this tactic will still mess up any other race with multiple robotic limbs/organs just as easily. My point is: if your issue is getting EMP'd through a wall at obnoxious range, nerf the EMP source (smaller range, higher cost, etc), not buff the victim's resistance to it. That's like arguing that people should be more resistant to bullets if they don't want to get shot rather than making it simply harder to shoot them in the first place. It is outright absurd to say IPCs should be more resistant to EMPs, when the entire purpose of EMPs is to ruin electronics. I could see this argument as valid if the IPC was supposed to be some sort of clockwork golem which is 100% mechanical, but IPCs are definitely electronic.

**Comment by Ziiro** [comment]
>My point is: if your issue is getting EMP'd through a wall at obnoxious range, nerf the EMP source (smaller range, higher cost, etc), not buff the victim's resistance to it. 

No. If you tweak with EMP balance for the sake of **one race**, you're going to fuck over the antags to have to deal with secborgs or other things that are interacted with by EMPs (weapons, cameras, etc). Why adjust that when the species is the only real problem in this equation?

In my opinion, the simple truth is: 

EMPing IPCs as is is miserable, shitty design. There is no counterplay. There is no preparation. There is no skill involved on behalf of the antag. **There is nothing you can do except not play the race.** One click and you're removed from the game with 15 minutes? 30 minutes? to return after people collect your parts and repair your internal damage - assuming they even notice or find you. I've been EMP flashlighted **multiple times** in crowded areas and no one notices, because it actually makes no sound or indication that it happened.

I understand death is common and fast in SS13, but EMP doesn't even make for a good story or entertaining way to die.

**Comment by AdamElTablawy** [comment]
EMPs are not rare, for whoever's arguing that. Several antags have EMP abilities built right in/able to be purchased nigh immediately, and otherwise they're fairly easily produced. It's also a ranged instant kill with absolutely no warning, and usually no indiciation where it came from.

An instant kill is not a good race downside. There should at least be some level of manageability. Considering EMPs stun them and kill them in two hits, not one, with this change, not to mention potential delimbing, there's really no issue. 

They should be vulnerable to things, yes. However, considering IPCs generally suck at combat because of how fragile they are and their inability to use chemicals, it's not as though making them die in two hits to EMPs will make them more powerful. It still kills them. It still stuns them, too, so you can kill them personally if you don't want to use another EMP charge. All it does is mean that they can gurgle 'HELP' if their headset is, for some reason, still working, before they fall over dead again.

As for the argument that 'they should die to EMP's because they're synthetic:'

1. they still die to EMPs anyway

2. if a cyborg isn't instant killed by a half assed EMP there's no reason the Captain should get immediately destroyed by one while he's in the bridge surrounded by security officers


Also, it makes setting off an EMP to drain security officer's weapons or shut down a cyborg attacking you much more viable because you don't run the risk of getting pinged for murderbone because three schmuck IPCs got killed by it because they were in an entirely different room.

**Comment by ZN23XZ** [comment]
I think IPC players who hate EMP are overestimating how "good" it will feel to easily die as opposed to instantly. As someone who's 2 main toons both die to EMP (my human more than my IPC...occupational hazard working sec) and someone who frequently gets wordlessly killed for trying to RP in sec, I can confirm that dying instantly doesn't feel any worse than dying easily.
I know many IPC EMP haters would argue anything is better than instant, but you are a fool if you think dying to 2 EMPs or 1 EMP blowing your limbs off and making you helpless will make you feel better about it.

**Comment by Ziiro** [comment]
Hello everyone today we're going to talk about numbers, I'll be your caveman coder host Ziiro. I've changed a few things since the initial PR, now that I'm not as tired and had time to actually properly test numbers in actual play.

* This implementation of my tweak now deals **10** base burn damage to each limb per EMP. When you factor in that IPCs take 30% more burn damage, this is a total of 13 per limb. Or, a total of 156 (Assuming nothing gets dismembered)
* With a 15% rolled 10 times for each EMP, chances are very good they'll lose **at least** one limb. Probably more! Keep in mind, too, that's per limb. So they may not lose the diceroll on their hand, but they can still lose their arm. Well. Hand's gone anyway. If I knew statistics better, I'd work out the curve for this. That being said, I failed algebra twice. So you'll have to ask someone else.
* The stun is 30 seconds long. It's a stun. This is a game where you live and die based on being stunned *at all*. If an antag cannot capitalize or do what they want during a 30 second stun where the victim doesn't have access to their radio, they don't deserve that kill.

**Comment by DesolateG** [comment]
IPC parts should rust when not on a living IPC. IPC chassis should rapidly deteriorate as well when an IPC bluescreens.

**Comment by shazbot194** [comment]
Can you double check your math, as I'm rather sure IPCs have a 150% burn modifier, which would put the damage to 195 burn.

I'm also rather sure you cannot slap an IPC'S arm with an emagged and have it explode, as that is a very old bay feature. However, if I am wrong and you can, that seems rather unbalanced.

**Comment by Rurik123** [comment]
In the end, if you use an emp with the intent to kill a IPC, it is still going to work with nothing more than a modicum of additional effort on your end. I think that much has been confirmed already. But I think some are perhaps missing a valuable upside to this. 

Allowing EMP's to nonlethally delay and stun IPC's opens up a whole new use for a EMP. It can be used where you want to capture/steal from one, without killing it, while still retaining the lethality when wanting one dead. It would become effectively two in one, allowing more use in scenarios where killing a IPC is not desirable over simply stunning it. In other words, it opens up another use for the EMP while retaining is deadliness, with the additional positive of making those who want one click kill mechanics to be gone happy. Perhaps I am missing something, but I fail to see the downside to this.

**Comment by Ziiro** [comment]
@shazbot194 

The numbers I was giving there were done in-game, using a cyborg analyzer after flashing them with an EMP flashlight.

**Comment by FalseIncarnate** [comment]
IPCs have a brute and burn mod of 2.5, and robotic limbs have a brute/burn mod of 0.6
 - This means damage taken is multiplied by 2.5, then multiplied by 0.6

For the sake of nice clean math, that means 10 damage is multiplied by 2.5, becoming 25 damage. This 25 is then multiplied by 0.6, becoming 15. Effective increase: 150%, as Shazbot correctly stated.

https://github.com/ParadiseSS13/Paradise/blob/master/code/modules/mob/living/carbon/human/human_defense.dm#L165 handles an emag being used on any human mob (which IPCs qualify as) to sabotage a robotic limb (which all IPC limbs are).

https://github.com/ParadiseSS13/Paradise/blob/master/code/game/objects/items/robot/robot_parts.dm#L423 handles an emag being used on a robotic limb fresh from the printer, for those curious.

**Comment by Ziiro** [comment]
>Effective increase: 150%, as Shazbot correctly stated.



chrome_2018-03-23_13-33-07

dreamseeker_2018-03-23_13-34-12

Note that the one limb that was *not* lost (man, that dice roll) has 13 burn damage. 30% more than 10.

**Comment by FalseIncarnate** [comment]
That seems like something got changed elsewhere then, since that's not how that math should line up... and looking into this it gets weirder.

Apparently robotic limbs actually modify brute/burn damage received by 0.66, so something isn't being accounted for somewhere...

**Comment by shazbot194** [comment]
I have to say, after looking over everything, I can't say I like this PR a lot. The ability to loose most if not all your limbs isn't really any better then being dead if you loose a leg and can't walk, or both arms and can't do anything, or even your head and now are blind. This means that one EMP can still do more then enough damage to disable an IPC enough to basically put them into a crit state instead of death. While loosing out on probably the only useful chem isn't really a great winner in my books, without the dismemberment, I would be more then happy with the changes however.

**Comment by Ziiro** [comment]
I could lower the chance of the dismemberment. It's balanced on the other codebase because there are less limbs to roll - here there are more, and I could probably get away with less.

It's still a diceroll, however.

Also, I do have a solution for the perma-crit problem:

```
/datum/species/ipc/spec_life(mob/living/carbon/human/H)
	. = ..()
	if(H.health <= HEALTH_THRESHOLD_CRIT && H.stat != DEAD) // So they die eventually instead of being stuck in crit limbo.
		H.adjustFireLoss(6) // After bodypart_robotic resistance this is ~2/second
		if(prob(5))
			to_chat(H, "<span class='warning'>Alert: Internal temperature regulation systems offline; thermal damage sustained. Shutdown imminent.</span>")
			H.visible_message("[H]'s cooling system fans stutter and stall. There is a faint, yet rapid beeping coming from inside their chassis.")
```

tldr: If they are in that crit limbo, it deals burn damage until they die.

**Comment by DesolateG** [comment]
"I could lower the chance of the dismemberment. It's balanced on the other codebase because there are less limbs to roll - here there are more, and I could probably get away with less."

So you're wanting to buff an already relatively balanced niche karma race even further?

**Comment by Ziiro** [comment]
I was actually referring to the way I have them on Oracle. /tg/ med has 6(5 that can be dismembered) limbs. So a 20% roll makes sense. Paradise has 12 limbs (10 that can be dismembered), so I made it 15%. Which, it could probably be 10%.

**Comment by granodd** [comment]
>So you're wanting to buff an already relatively balanced niche karma race even **_further?_**

I sure do enjoy buzzwords. I'm not sure how many people who say this truly believe it, or if there might be some ulterior motive behind it, but I'll respond with the assumption that you're all being truthful to your words. 

For one: "buff an already balanced race even **_further_**", I can safely assume that this is in relation to the race's "strengths" and not any buffs they've recently had, because they have had _no significant buffs_ within the passed year. Not as far as I can recall at least. In fact, they've had minor nerfs to their QoL. More specifically the change to how they recharge. 

Granted, the first one made sense, having them have to do a series of action-bar queues to drain power from an APC rather than just grab intent and then clicking a thousand times to completely drain an APC. Though the followup changes to how they recharged reduced that quality, first making their recharger something that can be damaged with EMP, and then later moving that recharger into their right arm on a species whose limbs are very fickle about staying attached to their body. In the end, it can be fixed with a trip to robotics and a printing of an arm and recharger, putting it more on-par with visiting doctors for fixing, though the point still stands that it was an overall decrease in their original QoL.

As for their strengths.
>• Immunity to two major damage types (oxy/toxins) and two minor (rad/clone)

When it comes to how often damage types pop up, the way I see it is: brute > burn > toxin >= oxy > rads > clone. Being immune to clone in the few cases that it _ever_ pops up is nice, while being immune to rads would be a bit more meaningful if rads were more prevalent, like say, if there were more rad storms than what already exists. 

Toxins usually only happen if someone is actively trying to poison you, or you're dumb (drink too much, get into a fight with a spider, etc.), and on that former one, not many people I see go for the "actively poison" when things like stuns and brute/burn are much easier to get and more reliable in killing your target. Oxy-loss outside of low blood can be avoided with the assistance of internals, or in the case of a punctured lung and low blood, salbutamol until you can get it fixed by somebody.

>• Immune to viruses

I'm going to go ahead and say this is a con. Reasoning being: super-healing viruses from virology have become much more common in recent months after the discovery that you can make three beneficial viruses and slap them into a single pill to fill up all of the virus slots that a single carbon can have (and these healing viruses can be pretty potent in deciding the difference between life or death while effectively negating a lot of the weaknesses that organics usually have, like oxy-loss or toxins). In addition, 90% of the viruses that pop up from a Level 7 biohazard are a joke. They're either something you can cure with orange juice, mannitol, or in the off chance it's magnetitus (which is more of an annoyance than a hindrance, and probably why so many people don't go to get it cured), iron.

 >• Immunity to pain, and thus higher brute/burn weakness

Immunity to pain I've come to respect a lot more, given how obnoxious it can be when low on health as an organic. The justification of the brute and burn here does make sense, though I must say, not in the current modifiers that it's at. I would argue that while EMP is an IPC's strongest weakness, their brute and burn modifier is the close second. I can't think of any race who has a weakness to both of these damages, let alone with modifiers so high that they feel more like people made of plastic and loosely fitted wires rather than actual robots. If their brute/burn modifier was more on par with other races that had similar modifiers (vox at 20% and greys at 25% with brute), then that would make them feel a bit less like plastic and perhaps a bit more like aluminium. Their limbs pop off like legos, anyhow, so whacking one with a toolbox will likely end in their demise if they don't have an appropriate tool to fight back with, right after you knock off both their hands and maybe an arm.

>• Immune to standard chems.

Pro and con, I'll agree. On one hand, no nice patches or whatever ( though healing patches on an IPC wouldn't make any sense anyhow). On the other, people with memechem syringe guns don't do anything to you as the syringes just plink right off. They are, however still susceptible like everyone else to lube+FLC3 memes, but I suppose it's fair that everyone must suffer through that.

>• Can be revived through cable coil and/or welder without the need of SR or rotting

No rotting is very nice, along with the fact that IPCs can be revived basically anywhere that has those tools. However, some changes need to be made to alert them to _when_ someone does this, so they don't just stand back up and go into the oh too common "catatonic state" that you usually find IPCs in. Though I will note in addition to their ease of access to being repaired in surgery, is also their ease of access of being debrained by anyone with a multitool, screwdriver, and crowbar, which is made even easier given that EMPs kill them instantly and silently (in the case of a flashlight). The same can happen to organics (debraining), albeit with questionably more difficulty (you're likely not going to kill someone quietly in an easy fashion, and sleepy pens take a bit more skill and timing to use, not to mention being a hefty 8 TC). But if you're prepared enough (or paranoid enough), a pre-scan can combat that. And though pre-scanning is very gamy, IPCs don't have access to that "luxury".

>• Cannot receive genetic mutations

I'll call this one a con-leaning neutral, based on the fact that you're unlikely to get mutations unless you're exposed to radiation, which as I noted before, would be more significant if radiation was more prevalent. 90% of the mutations that you see anyhow in a common round are beneficial ones from genetic memes, while a lot of the negative ones are easily fixed with only a single unit of mutadone or a clean SR injection.

Also, @Ziiro the perpetual crit issue can be fixed by typing "succumb" as opposed to a forced decline towards death, even as neat as the whole "cooling systems offline" thing sounds.

**Comment by Ziiro** [comment]
The problem I have with the succumb solution is it's just not intuitive. I remember even after I unlocked IPC. It wasn't until like 20 more hours in that someone told me to do that.

**Comment by shazbot194** [comment]
Somehow, no matter how I try to use "succumb", it never works, but I feel like that is a problem with me, and not with the code.

**Comment by davidchan13** [comment]
@shazbot194 You have to be in crit, like when you're getting messages to 'STOP THE PAIN!' and 'YOU FEEL LIKE YOU COULD DIE!', for IPCs is basically when you collapse and can't move or talk.

**Comment by granodd** [comment]
@davidchan13 I've always used it when I drop down into the state where I can't hear/see/do anything. It is a thing that IPC players should learn, though I will agree that it's an issue of being stuck in hellcrit when you don't know about the succumb verb. I however don't think it should be an automatic thing, personally, as I try to wait for someone to find my body for a while before I think of typing succumb. Perhaps give a prompt when you fall into hard crit of "Enter Emergency Shutdown (Death)? Y/N" @Ziiro

**Comment by sothangel** [comment]
I don't really see the need for most of these changes honestly. Sec can already use tasers and disablers on them for detaining. I'd rather see more chemicals added to interact with IPCs, or expanding the list that they interact with rather than just having them interact with nothing.

**Comment by DarkPyrolord** [comment]
Alright, after internal discussion it has been decided that this PR will be closed.

As of right now, IPCs are the least of our worries when it comes to species balancing as they are really in the most detailed and unique spot compared to most others, which either require buffs or nerfs to bring them all to a good baseline, this is a huge buff for IPCs and at this point this is not on the table.

Also by the way, #8783 is still on the table by all means and I am for it really, once things are fixed up I'll be happy to approve that one for merging, but for now, sorry about this one.

**Pull request closed without merging**